pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -3194,6 +3216,19 @@ func validateRegistrationEntry(entry *common.RegistrationEntry) error {
 		return sqlError.New("invalid registration entry: missing selector list")
 	}
 
+	// In case of StoreSvid is set, all entries 'must' be the same type,
+	// it is done to avoid users to mix selectors from different platforms in
+	// entries with storable SVIDs
+	if entry.StoreSvid {
+		// Selectors must never be empty
+		tpe := entry.Selectors[0].Type
```

Comment:
Can we use the equalSelectorTypes helper here?

Reply:
It is not possible, the selector we use here is from `proto common` (https://github.com/spiffe/spire/blob/main/proto/spire/common/common.pb.go#L340)
But equalSelectorTypes is done for `sqlstore/models` (https://github.com/spiffe/spire/blob/main/pkg/server/datastore/sqlstore/models.go#L80)

Reply:
Ah ok.